external/storm-kafka/src/jvm/org/apache/storm/kafka/DynamicBrokersReader.java

```diff
@@ -110,7 +110,7 @@ private int getNumPartitions(String topic) {
             List<String> children = _curator.getChildren().forPath(topicBrokersPath);
             return children.size();
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            LOG.error(e.getMessage(), e);
```

Comment:
Which value do you think this method returns when exception is caught, which action caller should take? Without proper action for failing back, it is better to throw exception and fail-fast.

Reply:
And compilation fails on the change. Please see the result of Travis build.

Reply:
@HeartSaVioR 
When exception is caught,we should only print the error messages,so that users can be aware of there are some problems in their cluster such as one role of zk was down.

Reply:
@HeartSaVioR 
For example,one  role of zk is down in users cluster,storm-kafka will throw a exception,even storm also can connect another zk role and works.I will check the  result of Travis build.

Reply:
When RuntimeException is propagated to call hierarchy, some place will leave a error message. (it is guaranteed when worker is crashing) 

Btw, I guess you may need to think and try to answer my questions above by yourself.
Please note that core principal of Storm is 'fail-fast', because in many case things are not restorable and retryable, and we had to kill the process to prevent process to go with unexpected behavior.

Reply:
That may be viable and possible, but you may need to add retry logic rather than just removing throw statement.

Reply:
I think it doesn't affect 'fail-fast',I just want to guarantee users' Topology can work normally.

Reply:
@HeartSaVioR 
I mean topology should  work even we  can't connect one of the zkservers,because the data is shared  among zkservers. But if we throw a exception,storm will not attempt to connect other zkserver,and exit.